rpc/item.c

```diff
@@ -1594,6 +1667,24 @@ M0_INTERNAL void m0_rpc_item_cache_clear(struct m0_rpc_item_cache *ic)
 	}
 }
 
+M0_INTERNAL void m0_rpc_item_xid_list_init(struct m0_rpc_session *session)
+{
+	M0_PRE(m0_rpc_machine_is_locked(session->s_conn->c_rpc_machine));
+	xidl_tlist_init(&session->s_xid_list);
+}
+
+M0_INTERNAL void m0_rpc_item_xid_list_fini(struct m0_rpc_session *session)
+{
+	struct m0_rpc_item *item;
+
+	M0_PRE(m0_rpc_machine_is_locked(session->s_conn->c_rpc_machine));
+	m0_tl_for(xidl, &session->s_xid_list, item) {
+		M0_LOG(M0_ERROR, "item="ITEM_FMT" ri_sm.sm_state=%d",
+		       ITEM_ARG(item), item->ri_sm.sm_state);
```

Comment:
Is the link removed from the list and finalized?

Reply:
It is. In `m0_rpc_item_fini()` -> `rpc_item_xid_unassign()`. Lifetime of rpc session should be more than a lifetime of all rpc items in it, so when the session is finalised any item in the list is a bug.